src/App.vue

```diff
@@ -1,18 +1,55 @@
 <template>
-    <img alt="Vue logo" src="./assets/logo.png" />
-    <HelloWorld msg="Welcome to Your Vue.js + TypeScript App" />
+    <main>
+        <Level :level="level" />
+        <Timer :time="time" />
+        <Buffer :slots="slots" :codes="codes" />
+        <Matrix :size="size" :matrix="matrix" />
+    </main>
 </template>
 
 <script lang="ts">
-import { defineComponent } from 'vue';
-import HelloWorld from './components/HelloWorld.vue';
+    import {defineComponent} from "vue";
+    import Level from "./components/Level.vue";
+    import Timer from "./components/Timer.vue";
+    import Buffer from "./components/Buffer.vue";
+    import Matrix from "./components/Matrix.vue";
 
-export default defineComponent({
-    name: 'App',
-    components: {
-        HelloWorld
-    }
-});
+    export default defineComponent({
+        components: {
+            Level,
+            Timer,
+            Buffer,
+            Matrix
+        },
+        setup() {
+            const time = new Date();
+            time.setSeconds(time.getSeconds()+60);
+
+            return {
+                level: 5,
+                time,
+                slots: ["99", "E7", "AD", ""],
+                codes: [
+                    {
+                        code: ["", "E7", "AD", "BD"],
```

Comment:
What does it do when the first is empty? I though it does not matter when the sequence is started? Does it have to align with the buffer?

Reply:
That's used as an offset.